plugins/validate/jsonschema.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The action plugin file for cli_parse
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Ganesh Nalawade (@ganeshrn)
+    validate: jsonschema
+    short_description: Define configurable options for jsonschema validate plugin
+    description:
+    - This plugin documentation provides the configurable options that can be passed
+      to the validate plugins when I(ansible.utils.json) is used as a value for
+      engine option.
+    version_added: 1.0.0
+    options:
+      draft:
+        description:
+        - This option provides the jsonschema specification that should be used
+          for the validating the data. The C(criteria) option in the C(validate)
+          plugin should follow the specifiaction as mentined by this option
+        default: draft7
+        choices:
+        - draft3
+        - draft4
+        - draft6
+        - draft7
+        env:
+        - name: ANSIBLE_VALIDATE_JSONSCHEMA_DRAFT
+        vars:
+        - name: ansible_validate_jsonschema_draft
+Notes:
+- The value of C(data) option should be either of type I(dict) or I(strings) which should be
+  a valid I(dict) when read in python.
+- The value of C(criteria) should be I(list) of I(dict) or I(list) of I(strings) and each
+  I(string) within the I(list) entry should be a valid I(dict) when read in python.
+"""
+
+import json
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils.six import string_types
+
+from ansible_collections.ansible.utils.plugins.validate import ValidateBase
+
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    to_list,
+)
+
+try:
+    import jsonschema
+
+    HAS_JSONSCHEMA = True
+except ImportError:
+    HAS_JSONSCHEMA = False
+
+
+def to_path(fpath):
+    return ".".join(str(index) for index in fpath)
+
+
+def json_path(absolute_path):
+    path = "$"
+    for elem in absolute_path:
+        if isinstance(elem, int):
+            path += "[" + str(elem) + "]"
+        else:
+            path += "." + elem
+    return path
+
+
+class Validate(ValidateBase):
+    @staticmethod
+    def _check_reqs():
+        """Check the prerequisites are installed for jsonschema
+
+        :return dict: A dict with a list of errors
+        """
+        errors = []
+        if not HAS_JSONSCHEMA:
+            errors.append(missing_required_lib("jsonschema"))
+
+        return errors
+
+    def _check_args(self):
+        """Ensure specific args are set
+
+        :return: A dict with a list of errors
+        :rtype: dict
+        """
+        errors = []
+        try:
+            if isinstance(self._data, dict):
+                self._data = json.loads(json.dumps(self._data))
```

Comment:
doing this to clear out any Ansible types?

Reply:
Yes that's right